From 6d7f8f77baa764e55409a12c9a8d8261a770d283 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Sun, 1 Dec 2013 12:54:55 +0100
Subject: [PATCH 260/277] MEDIUM: peers: delay appctx initialization

Now that the session handler can automatically initialize the appctx,
let's not do it in peers_accept() anymore.
---
 src/peers.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/peers.c b/src/peers.c
index 234c4b9..0d06505 100644
--- a/src/peers.c
+++ b/src/peers.c
@@ -1089,14 +1089,8 @@ static void peer_session_forceshutdown(struct session * session)
  */
 int peer_accept(struct session *s)
 {
-	struct appctx *appctx;
-
 	s->target = &peer_applet.obj_type;
-	appctx = stream_int_register_handler(&s->si[1], objt_applet(s->target));
-	if (!appctx)
-		return -1;
-	appctx->st0 = PEER_SESS_ST_ACCEPT;
-	appctx->ctx.peers.ptr = s;
+	/* no need to initialize the applet, it will start with st0=st1 = 0 */
 
 	tv_zero(&s->logs.tv_request);
 	s->logs.t_queue = 0;
-- 
1.8.1.5

